src/puppet.js

```diff
@@ -387,51 +575,37 @@
       }
     });
 
-    // choose queuing engine
-    if(options.localVersionPath){
-      if(!options.remoteVersionPath){
-        //just versioning
-        this.queue = new JSONPatchQueueSynchronous(options.localVersionPath, this.validateAndApplySequence.bind(this), options.purity);
-      } else {
-        // double versioning or OT
-        this.queue = options.ot ?
-          new JSONPatchOTAgent(JSONPatchOT.transform, [options.localVersionPath, options.remoteVersionPath], this.validateAndApplySequence.bind(this), options.purity) :
-          new JSONPatchQueue([options.localVersionPath, options.remoteVersionPath], this.validateAndApplySequence.bind(this), options.purity); // full or noop OT
-      }
-    } else {
-      // no queue - just api
-      this.queue = new NoQueue(this.validateAndApplySequence.bind(this));
-    }
-
     this.ignoreCache = {};
     this.ignoreAdd = options.ignoreAdd || null; //undefined, null or regexp (tested against JSON Pointer in JSON Patch)
-    this.pingInterval = options.pingInterval || false;
 
     //usage:
     //puppet.ignoreAdd = null;  //undefined or null means that all properties added on client will be sent to remote
     //puppet.ignoreAdd = /./; //ignore all the "add" operations
     //puppet.ignoreAdd = /\/\$.+/; //ignore the "add" operations of properties that start with $
     //puppet.ignoreAdd = /\/_.+/; //ignore the "add" operations of properties that start with _
 
-    var onDataReady = options.callback;
-    var puppet = this;
-    this.network.establish(function bootstrap(responseText){
-      var json = JSON.parse(responseText);
-      var bigPatch = [{ op: "replace", path: "", value: json }];
-      puppet.validateAndApplySequence(puppet.obj, bigPatch);
-
-      if (puppet.debug) {
-        puppet.remoteObj = responseText; // JSON.parse(JSON.stringify(puppet.obj));
-      }
-
-      recursiveMarkObjProperties(puppet.obj);
-      puppet.observe();
-      if (onDataReady) {
-        onDataReady.call(puppet, puppet.obj);
+    this.onDataReady = options.callback;
+
+    this._createQueue = function() {
```

Comment:
How it is easier to write `this._createQueue` instead of `createQueue`, or even `` -as it could be executed inline?

Plus it makes API messy, it make another method public, and as you said, it's harder to remove things from API rather than add them.
It makes developer wonder, whether it is used ot of this scope, or by any other module.

If it is local you do not need to remember it once you have finished reading constructor.